mathoxide-lib/src/array.rs

```diff
@@ -59,6 +59,43 @@ where
     pub fn storage_size(&self) -> usize {
         self.storage.storage_len().expect("Cannot get array length")
     }
+
+    pub fn get<ListType: AsRef<[usize]>>(
+        &self,
+        idx: ListType,
+    ) -> Array<StorageType, ContiguousView> {
+        let offset = match self.view.try_translate(&idx) {
+            Ok(u) => u,
+            Err(_) => panic!(
+                "Index {:?} is not valid for a view with shape {:?}",
+                idx.as_ref(),
+                self.shape()
+            ),
+        };
```

Comment:
Maybe do something like `self.view.try_translate(&idx).unwrap_or_else(|_| panic!("ah: {}", "nuts"))` to avoid the `match` expression?